swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md

```diff
@@ -0,0 +1,501 @@
+│   │		└── model.bal
+│   ├── Ballerina.toml
+│   └── main.bal
+
+```
+And adds the following configuration to the `Ballerina.toml` file.
+
+```toml
+[persist]
+datastore = "inmemory"
+module = "rainier.store"
+```
+
+
+These configurations are referred to when generating client objects for the data model. The recommendation is not to change these values. If you want to change it, remove these configurations from the `Ballerina.toml` file and reinitialize the `bal persist`.
+
+The next step is to define your data model in the schema file at `persist/model.bal`.
+
+## Model your data
+
+The data model can be defined in the schema file inside the `persist` directory. You can use the empty file added by the `persist init` command for this.
+
+For more information on defining a data model, refer to the [Data Model Definition](learn/persist-model/) documentation. In order to simplify this demonstration, only the `Employee` entity is added to the schema file. You can add as many as you want for your application and add relationships between entities.
+Once defined, the schema file looks like this.
+
+```ballerina
+import ballerina/persist as _;
+import ballerina/time;
+
+public type Employee record {|
+   readonly string employee_id;
+   string first_name;
+   string last_name;
+   string email;
+   string phone;
+   time:Date hire_date;
+   string? manager_id;
+   string job_title;
+|};
+```
+
+This model is used to set up the underline datastore (e.g., set up the tables in the underlying database) and also as the base for the generated client APIs
+
+## Generate the client object, types, and scripts
+
+Now you can generate the client objects, types, and SQL scripts for your model by running the command in your terminal,
+
+```
+$ bal persist generate
+```
+
+This changes the Ballerina package as follows.
+
+```
+├── rainier
+│   ├── generated
+│   │		└── db
+│   │		│	├── persist_client.bal
+│   │		│	└── persist_types.bal
+│   ├── persist
+│   │		└── model.bal
+│   ├── Ballerina.toml
+│   └── main.bal
+    
+```
+
+The `persist generate` will parse the `persist/model.bal` definition file and generate the following.
+
+|         File          |                                         Description                                         |
+|:---------------------:|:-------------------------------------------------------------------------------------------:|
+| `persist_client.bal`  |               This is the client that is used to persist and retrieve data..                |
+|  `persist_types.bal`  |       This contains the record types that will be used to persist and retrieve data.        |
+
+
+
+> Note: All of the above auto-generated bal files should not be modified.
+
+> Note: If you use other data stores like MySQL or Google Sheets, additional files are created to handle runtime configurations and scripts to set up the database/worksheet. Since we are using `in-memory`, we don’t need any configuration and pre-setting up to run the application.
```

Comment:
```suggestion
> Note: If you use other data stores like MySQL or Google Sheets, additional files are created to handle runtime configurations and scripts to set up the database/worksheet. Since `in-memory` is used, you don’t need any configuration and pre setting up to run the application.
```
Can we combine these 2 notes?